packages/ai/src/providers/google/index.ts

```diff
@@ -80,7 +80,10 @@ const createChatCompletion: ICreateChatCompletion = async (body) => {
         return modelInstance
           .generateContent({ contents })
           .catch((err: Error) => {
-            retry(err);
+            // TODO: Replace with instanceof checks when the Gemini SDK exports errors
+            if (err.message.includes("[429 Too Many Requests]")) {
```

Comment:
Oh we don't retry for intermediate 5xx ? Or is the Gemini sdk handling it ?

Reply:
Good point. I did not see those errors while testing the v1beta endpoint (which does not mean they will not happen). The JS SDK is not handling them (which is why we need this hack). Might be better to use the REST endpoint directly and react based on error payload/status code

Reply:
yup .. moving to fetch method can even make these handling a bit more clean as well. 

cc @KaustubhKumar05